server/tserver/src/main/java/org/apache/accumulo/tserver/compaction/EverythingCompactionStrategy.java

```diff
@@ -27,7 +27,7 @@
 
   @Override
   public boolean shouldCompact(MajorCompactionRequest request) throws IOException {
-    return request.getFiles().size() > 0;
+      return true; // ACCUMULO-3645 compact for empty files too
```

Comment:
Rather than change the default compaction strategy, it'd probably be better to introduce a new compaction strategy, which returns true, instead of implementing the optimized behavior.